File: src/main/java/dk/cngroup/fetch/controller/RestExceptionHandler.java
package dk.cngroup.fetch.controller;

import lombok.extern.slf4j.Slf4j;
import org.springframework.dao.EmptyResultDataAccessException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
import static org.springframework.http.HttpStatus.NOT_FOUND;

@RestControllerAdvice
@Slf4j
public class RestExceptionHandler {

    @ExceptionHandler(EmptyResultDataAccessException.class)
    @ResponseStatus(NOT_FOUND)
    public String handleNotFoundException(Exception e) {
        return e.getMessage();
    }

    @ExceptionHandler(Throwable.class)
    @ResponseStatus(INTERNAL_SERVER_ERROR)
    public String handleAnyException(final Throwable t) {
        log.error("Internal server error", t);
        return t.getMessage();
    }
}

File: src/main/java/dk/cngroup/wishlist/controller/RestExceptionHandler.kt
package dk.cngroup.wishlist.controller

import mu.KotlinLogging
import org.springframework.dao.EmptyResultDataAccessException
import org.springframework.http.HttpStatus.*
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class RestExceptionHandler {

    @ExceptionHandler(EmptyResultDataAccessException::class)
    @ResponseStatus(NOT_FOUND)
    fun handleNotFoundException(e: Exception): String? {
        return e.message
    }

    @ExceptionHandler(Throwable::class)
    @ResponseStatus(INTERNAL_SERVER_ERROR)
    fun handleAnyException(t: Throwable): String? {
        logger.error("Internal server error", t)
        return t.message
    }
}

private val logger = KotlinLogging.logger {}